### PR TITLE
Automated cherry pick of #1729: Add component for the eck controller version.

### PR DESCRIPTION
**Pull request by doublek**
Cherry pick of #1729 on release-v1.24.

#1729: Add component for the eck controller version.

### CODE REVIEW

File: config/enterprise_versions.yml [hunk 1]
```diff
@@ -114,6 +114,8 @@ components:
   # quay.io/tigera so all enterprise images come from the same repository and org.
   elasticsearch-operator:
     image: tigera/eck-operator
+    version: release-calient-v3.11
```
Comment (doublek) [inline config/enterprise_versions.yml:117]: Leaving this until we publish an image.

**Review by bcreane — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**